### DISCUSSION TITLE
API for roms and metadata

### DISCUSSION THREAD

**Discussion by IDmedia** [emmercm/igir#1029] [General]
Hi,

I've wanted to sort my rom collection for a very, very long time and create a proper 1G1R set as I currently have games accross 53*4TB drives. Having everything stored in cold storage and unable to browse through or play it doesn't make much sense. It also makes no sense having every rom in every language possible, hence 1G1R... I'm never going to sort everything by hand and that's how I came across this amazing project (igir) and also retool from unexpectedpanda.

I've used systems like sonarr and radarr which are able to download/scan, move, sort and scrape data for movies and tvshows and they do an amazing job. Strangly enough there doesn't seem to exist a proper solution that does the same thing for roms/games/emulators. There exists a ton of frontends ans systems like launchbox, but I don't want to store all the data on the client and in general I was never happy with how the system looks or organizes the data. I'm not very interested in the downloading part, but I would love a simple webUI or API where I can see all platforms and what games I have and which I'm still missing. The killer feature would be that all you have to do is to drop roms into a "import" directory and it would get scanned, moved and metadata downloaded. This way the API could provide platform, game and emulation related information to frontends using a standarized that handles all the data properly and storing them on a server and not the client itself.

In this regard I'm still trying to figure out what system I should use. I love the simplicity if igir, but I'm not sure which tool does the best job in regards to 1G1R. It seems that retool is maintaining its own repo with information about clones while igir relies more on DAT files? If anyone has optionions about this I would love to hear them :)

**Comment by emmercm** [comment]
Hi @IDmedia, thank you for the thoughtful post!

RE: Sonarr and Radarr, https://github.com/rommapp/romm is probably the tool that most closely matches the experience you're looking for, thought I haven't used it yet myself. Maybe https://github.com/gaseous-project/gaseous-server as well.

RE: Retool and their custom DB- from my understanding, they're trying to fill in the gaps of some DAT groups such as Redump that don't publish parent/clone information. Igir has some [parent/clone inference](https://igir.io/dats/processing/#parentclone-inference) capability, but it will likely never be as good as Retool. But once you have a DAT with parent/clone information, Retool and Igir should have parity on 1G1R filtering and preferences.

I've thought about IGDB integration similar to Romm as a way to lowering the barrier to entry with Igir (DATs are fairly confusing to newcomers), but it's really low on my priority list.

I suspect that Igir isn't the right tool for the experience you're looking for. Igir, by intentional design, is stateless and never creates a DB of ROMs in your collection. It's really a tool for file organization, with some [report-generation functionality](https://igir.io/output/reporting/). I think the `igir report` command could get you some of what you want in identifying gaps in your collection, but it would take some work to merge the reports of so many hard drives into one.

All that said, I would love to hear feedback about functionality gaps you see in Igir! But I don't want to convince you that it's the right tool for you when it might not be.

**Reply by benjamfk** [reply]
From my understanding Igir and retool complement each other. While retool might be better for 1G1R because of their custom DB, it doesn't move files around, but just filteres the DAT. Igir on the other hand can take this DAT and run it agains a directory to sort to find and move the files to a output directory. It's ability to create reports is good enough to parse and use.

That's why I'm looking into writing an API that takes both of these applications and wraps them. The only thing the user should have to do it so put DAT files in a folder and it should automatically add it to the API. The user should be able to tweak what to include. Then it's just a matter of dropping games into the processing directory and let the API figure out how to organize it. Downloading metadata, videos and serving the data for a future frontend. The frontend should display the games like Steam or gaseous-server and let users download a single package/bundle with emulator and rom ready to play without having to configure the game.

**Comment by emmercm** [comment]
In addition to 1G1R, you might like "best of" lists such as the MAME [All Killer No Filler](http://forum.arcadecontrols.com/index.php/topic,149708.0.html) list to help cut down your collection.
